### DISCUSSION TITLE
Broadcom NetXtreme II BCM57810 10 Gigabit Ethernet support

### DISCUSSION THREAD

**Discussion by astekolshchikov** [ipxe/ipxe#925] [Q&A]
I know that Broadcom NetXtreme II BCM57810 10 Gigabit EthernetNIC  is not in hardware drivers list, 
but could anyone tell me - is there any evidence of adding support of this NIC?
Or can anyone point me to manual how to add driver to iPXE build?

NIC is PCI-e standalone dual-port SFP+ card, VID 14e4 PID 168e, using linux kernel driver named bnx2x.

Actually we're experiencing some troubles with this NIC, but only when using iPXE, 
so for now i cannot point that iPXE driver abscense is cause of these troubles.

**Comment by NiKiZe** [comment]
Drivers for iPXE is written based on hardware specification. "Programmers reference guide"
NetXtreme II code exists in bnx2 in iPXE as well, it might need more work (rewrite) to properly support more devices in that series. (The code is larger than usual iPXE drivers due to being based on Linux sources, rather than hardware specifications)

If you are booting in EFI mode then the driver from NIC or firmware is used, if that is causing issues you might want to try and update the firmware.

**Reply by astekolshchikov** [reply]
> If you are booting in EFI mode then the driver from NIC or firmware is used, if that is causing issues you might want to try and update the firmware.

Yes, we booting in EFI mode. Last actual NIC firmware installed.
Tried Ubuntu 20.04, Debian 11, Windows 2016 Server - all fine.But only when iPXE was booted, we have troubles. 

If system is booted into iPXE menu - then after some time waiting in iPXE menu 
any system boot fails due network connectivity issues.
From tftp/http server side it looks like ip address of iPXE client is not accessible.
IP of iPXE client cannot be pinged, arping also fails to resolve.
On switch side iPXE client port is up, but no MAC present.

Temporary workaround : manually set menu default to system that's what we need and timeout to 1.

**Reply by NiKiZe** [reply]
Let's try and keep to the issue of NIC not working in iPXE.
If you boot snponly.efi from that NIC, you should only see that interface in iPXE and using the NII or SNP driver, this is the driver provided by firmware. If that isn't working you should reach out to that firmware vendor.

OSs does not use the firmware driver so not relevant.

Sure iPXE could provide a native driver, that would require some development time, as well as the mentioned hardware documentation and hardware to test with. But this should also work just fine with the SNP/NII stack and that seem to be buggy.

**Reply by astekolshchikov** [reply]
Yes, i found that for NetXtreme II BCM57810 10 Gigabit Ethernet support iPXE using NII driver.
As you stated that " If that isn't working you should reach out to that firmware vendor." - sad, but presume there's no way make NIC working w/o issues.
